### PR TITLE
Admin Menu: add support for Jetpack Redirect urls

### PR DESCRIPTION
**Pull request by rcanepa**
#### Changes proposed in this Pull Request

* Added support for Jetpack Redirect URLs as per https://github.com/Automattic/jetpack/pull/20141#issuecomment-890866524.

#### Testing instructions

* Download this PR.
* Start Calypso with `yarn start`.
* Follow the rest of the instructions on https://github.com/Automattic/jetpack/pull/20141 making sure you tested them on `calypso.localhost:3000` and not on `wordpress.com`.

### CODE REVIEW

File: client/state/data-layer/wpcom/sites/admin-menu/index.js [hunk 1]
```diff
@@ -19,12 +19,15 @@ export const requestFetchAdminMenu = ( action ) =>
...
-const sanitizeUrl = ( url, wpAdminUrl ) => {
+const sanitizeUrl = ( url, wpAdminUrl, siteUrl ) => {
 	const isSafeInternalUrl = new RegExp( '^/' ).test( url );
 	// The replace function removes the protocol.
 	const isSafeWpAdminUrl = new RegExp( `^${ wpAdminUrl?.replace( /^https?:\/\//, '' ) }` ).test(
 		url?.replace( /^https?:\/\//, '' )
 	);
+	const isSafeJetpackRedirectUrl =
+		new RegExp( '^https://jetpack.com/redirect/' ) &&
+		url?.includes( `site=${ siteUrl.replace( /^https?:\/\//, '' ) }` );
```
Comment (rcanepa) [inline client/state/data-layer/wpcom/sites/admin-menu/index.js:28]: We consider a URL safe if:
1. It is Jetpack Redirect URL (starts with `https://jetpack.com/redirect/`).
2. Contains the URL of the current site in the `site` query parameter.

**Comment by samiff** [comment]
This PR is required for https://github.com/Automattic/jetpack/pull/20141 which has gone a bit stale.

I've tried pushing this one along, would someone be willing to review this PR?

CC: @rcanepa @mattgawarecki @cpapazoglou @keoshi 

As far as I can tell, once we land this PR we can then merge https://github.com/Automattic/jetpack/pull/20141 & https://github.com/Automattic/wp-calypso/pull/53884

---

Here is how I've tested this PR:

* Applied D63148-code to my sandbox.
* Checked out this branch and ran `yarn start`
* For WP Admin, I started on page: https://example_site_url.wordpress.com/wp-admin/edit.php?preferred-view=classic
* For Calypso, I started on page: http://calypso.localhost:3000/posts/my/example_site_url.wordpress.com
* In both cases, the `Upgrades > Plans` URL resolves to: https://wordpress.com/plans/example_site_url.wordpress.com?site=example_site_url.wordpress.com

Calypso After:

Markup on 2021-09-24 at 22:50:56

**Review by rcanepa — COMMENTED** [review]
LGTM. It definitely solves the issue (I can't approve the PR since I opened it).

**Comment by keoshi** [comment]
LGTM too!

**Comment by samiff** [comment]
After clearing up a bit of confusion in https://github.com/Automattic/jetpack/pull/20141#issuecomment-944666120 it's not clear if we'd need this anymore, closing.

**Pull request closed without merging**